README.adoc

```diff
@@ -85,70 +92,102 @@ Sign into the application with one of the following user credentials from the us
-
-The [hotspot=loginPage file=0]`welcome.html` and [hotspot=errorPage file=0]`error.html` pages that implement the login form and error page respectively are provided for you under the `src/main/webapp` directory. The login form in the [hotspot=loginPage file=0]`welcome.html` page uses the `j_security_check` action, which is defined by Java EE and available by default.
-
-Authorization determines whether a user can access a resource. To restrict access to authenticated users with `user` and `admin` roles only, define the [hotspot=ServletSecurity file=0]`@ServletSecurity` annotation with the [hotspot=HttpConstraint file=0]`@HttpConstraint` annotation and set the [hotspot=HttpConstraint file=0]`rolesAllowed` attribute to these two roles.
-
-The [hotspot=TransportGuarantee file=0]`transportGuarantee` attribute defines the constraint on the traffic between the client and the application. Set it to [hotspot=TransportGuarantee file=0]`CONFIDENTIAL` to enforce that all user data must be encrypted, which is why an HTTP connection from a browser will be switched to an HTTPS one.
-
-The SecurityContext interface provides programmatic access to Java EE Security API. Inject a SecurityContext instance into the [hotspot=HomeServlet file=0]`HomeServlet` class. The [hotspot=doGet file=0]`doGet()` method uses the [hotspot=CallerInRole file=0]`isCallerInRole()` method from SecurityContext API to check a user's role and then forwards the response to the appropriate page.
-
-The [hotspot=webxmlsecurity file=1]`src/main/webapp/WEB-INF/web.xml` file contains the rest of the security declaration for the application.
+The [hotspot=HomeServlet file=0]`HomeServlet` servlet is the entry point of the application. 
+To enable form authentication for the [hotspot=HomeServlet  file=0]`HomeServlet` class, 
+define the [hotspot=AuthenticationMechanism file=0]`@FormAuthenticationMechanismDefinition` annotation 
+and set its [hotspot=loginToContinue file=0]`loginToContinue` attribute with a 
+[hotspot=loginToContinue file=0]`@LoginToContinue` annotation defining [hotspot=loginPage file=0]`welcome.html` 
+as the login page and [hotspot=errorPage file=0]`error.html` as the error page.
+
+The [hotspot=loginPage file=0]`welcome.html` and [hotspot=errorPage file=0]`error.html` pages 
+that implement the login form and error page respectively are provided for you under the 
+`src/main/webapp` directory. The login form in the [hotspot=loginPage file=0]`welcome.html` page 
+uses the `j_security_check` action, which is defined by Jakarta EE and available by default.
```

Comment:
Update to:
"The [hotspot=loginPage file=0]`welcome.html` page implements the login form, and the [hotspot=errorPage file=0]`error.html` page implements the error page. Both pages are provided for you under the `src/main/webapp` directory. The login form in the [hotspot=loginPage file=0]`welcome.html` page uses the `j_security_check` action, which is defined by Jakarta EE and available by default."